File: v1/mac/src/docker-app/docker/docker/AnalyticEvent.swift
import Foundation

public enum AnalyticEvent {
    case AppLaunched, InstallAskForToken, InstallTokenIsValid, InstallAskForRootPassword, InstallShowWelcomePopup, AppRunning, Heartbeat
    
    func isCore() -> Bool {
        return self == Heartbeat
    }
    
    func text() -> String {
        let selfString = String(self)
        return String(selfString.characters.prefix(1)).lowercaseString + String(selfString.characters.dropFirst())
    }
}

File: win/src/Docker.Core/tracking/AnalyticEvent.cs
namespace Docker.Core.Tracking
{
    public class AnalyticEvent
    {
        public static readonly AnalyticEvent AppLaunched = new AnalyticEvent("appLaunched");
        public static readonly AnalyticEvent InstallShowWelcomePopup = new AnalyticEvent("installShowWelcomePopup");
        public static readonly AnalyticEvent AppRunning = new AnalyticEvent("appRunning");
        public static readonly AnalyticEvent Heartbeat = new AnalyticEvent("heartbeat");

        public string Name { get; }
        public bool IsCore => Heartbeat == this;

        internal AnalyticEvent(string name)
        {
            Name = name;
        }
    }
}
